include/odp/api/spec/schedule_types.h

```diff
@@ -97,7 +100,9 @@ extern "C" {
  * queue synchronization context. A thread holds the context until it
  * requests another event from the scheduler, which implicitly releases the
  * context. User may allow the scheduler to release the context earlier than
- * that by calling odp_schedule_release_ordered().
+ * that by calling odp_schedule_release_ordered(). However, this call is just
```

Comment:
It relates to the fact that application cannot use release_ordered() to solve the deadlock problem. It cannot do plain queue enq/deq after context release easily (without calling sechedule).

Reply:
Got it. That makes sense on `_enq()`. Though `_deq()` cannot maintain order since it can be done only on PLAIN queues. Or are we allowing scheduler implementation affect plain queue implementation.

Reply:
Ordering spec requires that enqueues done inside ordering context appear in destination queue in correct order. Destination queue can be either sched or plain type. So, e.g. if single thread is doing all dequeues on the destination queue, it will see all events come out from the plain in the right order.

Reply:
OK, I think we're both saying the same thing. What I mean is ordering can only happen on events retrieved via `odp_schedule()`, not ones coming from `odp_queue_deq()`.
Thread may `odp_queue_enq()` events coming from `odp_schedule()` to a `PLAIN` queue and those will be ordered.

Reply:
yes